ironfish/src/consensus/verifier.ts

```diff
@@ -364,33 +355,14 @@ export class Verifier {
     tx?: IDatabaseTransaction,
   ): Promise<VerificationResult> {
     return this.chain.db.withTransaction(tx, async (tx) => {
-      const processedSpends = new BufferSet()
-
       const previousNotesSize = block.header.noteSize
       Assert.isNotNull(previousNotesSize)
 
       for (const spend of block.spends()) {
-        if (processedSpends.has(spend.nullifier)) {
-          return { valid: false, reason: VerificationResultReason.DOUBLE_SPEND }
-        }
-
-        if (spend.size > previousNotesSize) {
-          return {
-            valid: false,
-            reason: VerificationResultReason.NOTE_COMMITMENT_SIZE_TOO_LARGE,
-          }
+        const verificationError = await this.verifySpend(spend, previousNotesSize, tx)
```

Comment:
Do we need this at all?

1. We're removing the double spend check
2. Could spend size checking against previousNotesSize could be done before we add the block?
3. Could spend commitments be checked before add?

Reply:
Yea that's a good point. The only case I can think of is if we would want to later allow spends to reference other spends in the same block. We're not doing that rn so yea I think it would be the same as checking before. Wdyt about that scenario though? Either way do you mind if I make that change separately / after this PR if we want to do it?

Reply:
This doesn't seem like a feature we'd want to have since we're a privacy coin and miners cannot order transactions as they can in non-privacy coins